shared/src/main/java/uk/gov/di/authentication/shared/services/mfa/MfaMethodsService.java

```diff
@@ -315,41 +319,71 @@ public Optional<MfaMigrationFailureReason> migrateSmsMfaToCredentialsTableForUse
-        Optional<String> maybePhoneNumber = Optional.ofNullable(userProfile.getPhoneNumber());
-        var userHasActiveDefaultMfaMethod =
-                Optional.ofNullable(userCredentials.getMfaMethods())
-                        .orElse(new ArrayList<>())
-                        .stream()
-                        .anyMatch(
-                                mfaMethod ->
-                                        PriorityIdentifier.valueOf(mfaMethod.getPriority())
-                                                        .equals(PriorityIdentifier.DEFAULT)
-                                                && mfaMethod.isMethodVerified()
-                                                && mfaMethod.isEnabled());
-        if (maybePhoneNumber.isEmpty() || userHasActiveDefaultMfaMethod) {
+            return Optional.of(MfaMigrationFailureReason.ALREADY_MIGRATED);
+
+        // TODO - AUT-2198 - This method obfuscates things a bit
+        //  It includes only enabled UserCredentials and verified phone numbers. What should we do
+        // if they aren't either of these?
+        var maybeNonMigratedMfaMethod =
+                getMfaMethodForNonMigratedUser(userProfile, userCredentials);
+
+        // Bail if no MFA methods to migrate
+        if (maybeNonMigratedMfaMethod.isEmpty()) {
             persistentService.setMfaMethodsMigrated(email, true);
             return Optional.empty();
+            // TODO - AUT-2198 - By doing this, we need to make sure that all new MFA methods are
+            // getting added the new way before running before running a bulk migration, otherwise
+            // we could end up with a situation where we think something is migrated but actually
+            // isn't.
+            // TODO - AUT-2198 - Maybe we shouldn't do this given the two TODOs below
         }
 
+        var nonMigratedMfaMethod = maybeNonMigratedMfaMethod.get();
+
+        return nonMigratedMfaMethod.method() instanceof AuthAppMfaDetail
+                ? migrateAuthAppToNewFormat(
+                        email,
+                        nonMigratedMfaMethod) // TODO - AUT-2198 - What should do for non-enabled
+                // auth app?
+                : migrateSmsToNewFormat(
+                        email,
+                        nonMigratedMfaMethod,
+                        userProfile); // TODO - AUT-2198 - What should do for non-verified phone?
+    }
+
+    private Optional<MfaMigrationFailureReason> migrateAuthAppToNewFormat(
+            String email, MfaMethodData mfaMethodData) {
+        var method = (AuthAppMfaDetail) mfaMethodData.method();
+        persistentService.migrateMfaMethodsToCredentialsTableForUser(
+                email,
+                MFAMethod.authAppMfaMethod(
+                        method.credential(),
+                        true,
+                        true,
+                        mfaMethodData.priorityIdentifier(),
+                        mfaMethodData.mfaIdentifier()));
```

Comment:
Here I think we'll want to generate a new mfa identifier as opposed to using the default hard-coded one we generate for a non-migrated method on read (same for sms)